frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/NpmMojo.java

```diff
@@ -84,6 +102,19 @@ protected boolean skipExecution() {
 
     @Override
     public synchronized void execute(FrontendPluginFactory factory) throws Exception {
+        String nodeVersion = getNodeVersion(workingDirectory, this.nodeVersion, this.nodeVersionFile, project.getArtifactId(), getFrontendMavenPluginVersion());
+
+        if (isNull(nodeVersion)) {
+            throw new LifecycleExecutionException("Node version could not be detected from a file and was not set");
+        }
+
+        if (!NodeVersionHelper.validateVersion(nodeVersion)) {
+            throw new LifecycleExecutionException("Node version (" + nodeVersion + ") is not valid. If you think it actually is, raise an issue");
+        }
+
+        String validNodeVersion = getDownloadableVersion(nodeVersion);
+        factory.loadNodeVersionManager(validNodeVersion);
```

Comment:
Note to myself: Does this actually install if it's missing? If it does, not sure we want that? Probably don't want to encourage this, should be installing once in the parent POM.

Reply:
It doesn't, it only sets the values of node and npm executable paths if version manager is available